google/api_core/retry/retry_unary.py

```diff
@@ -212,34 +151,18 @@ def retry_target(
         # pylint: disable=broad-except
         # This function explicitly must deal with broad exceptions.
         except Exception as exc:
-            if not predicate(exc):
-                raise
-            last_exc = exc
-            if on_error is not None:
-                on_error(exc)
-
-        if deadline is not None:
-            next_attempt_time = datetime_helpers.utcnow() + datetime.timedelta(
-                seconds=sleep
+            # defer to shared logic for handling errors
+            _retry_error_helper(
+                exc, deadline, sleep, error_list, predicate, on_error, exc_factory
             )
-            if deadline < next_attempt_time:
-                raise exceptions.RetryError(
-                    "Deadline of {:.1f}s exceeded while calling target function".format(
-                        timeout
-                    ),
-                    last_exc,
-                ) from last_exc
-
-        _LOGGER.debug(
-            "Retrying due to {}, sleeping {:.1f}s ...".format(last_exc, sleep)
-        )
-        time.sleep(sleep)
+            # if exception not raised, sleep before next attempt
+            time.sleep(sleep)
 
     raise ValueError("Sleep generator stopped yielding sleep values.")
 
 
-class Retry(object):
-    """Exponential retry decorator.
+class Retry(_BaseRetry):
```

Comment:
I made a couple other changes to the wording as well. It used to imply that the deadline was the "maximum time for the rpc", but it can't actually interrupt in-progress sync rpcs. So I describe it as a "window in which retries will be attempted" instead

Reply:
I like it.